extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,162 @@
+
+  // adds a dicomweb server configuration in the clients list
+  private _addConfiguration(givenConfig) {
+    const config = Object.assign({}, givenConfig);
+    config.qidoConfig = {
+      url: config.qidoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.wadoConfig = {
+      url: config.wadoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.qidoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.qidoConfig)
+      : new api.DICOMwebClient(config.qidoConfig);
+
+    config.wadoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.wadoConfig)
+      : new api.DICOMwebClient(config.wadoConfig);
+
+    config.qidoDicomWebClient.name = config.name;
+    config.wadoDicomWebClient.name = config.name;
+    this.clients.push(config);
+  }
+
+  public addConfiguration(params, query, config) {
+    if (
+      config.onConfiguration &&
+      typeof config.onConfiguration === 'function'
+    ) {
+      config = config.onConfiguration(config, {
+        params,
+        query,
+      });
+    }
+    this._addConfiguration(config);
+  }
+
+  public getAuthorizationHeader() {
+    const xhrRequestHeaders = {};
+    const authHeaders = this.userAuthenticationService.getAuthorizationHeader();
+    if (authHeaders && authHeaders.Authorization) {
+      xhrRequestHeaders.Authorization = authHeaders.Authorization;
+    }
+    return xhrRequestHeaders;
+  }
+
+  public generateWadoHeader(config) {
+    const authorizationHeader = this.getAuthorizationHeader();
+    //Generate accept header depending on config params
+    const formattedAcceptHeader = utils.generateAcceptHeader(
+      config.acceptHeader,
+      config.requestTransferSyntaxUID,
+      config.omitQuotationForMultipartRequest
+    );
+
+    return {
+      ...authorizationHeader,
+      Accept: formattedAcceptHeader,
+    };
+  }
+
+  // sets authorization headers before queries
+  public setQidoHeaders() {
+    this.clients.forEach(
+      client =>
+        (client.qidoDicomWebClient.headers = this.getAuthorizationHeader())
+    );
+  }
+
+  public setWadoHeaders(tipo = 1) {
+    if (tipo === 1) {
+      this.clients.forEach(
+        client =>
+          (client.wadoDicomWebClient.headers = this.generateWadoHeader(client))
+      );
+    } else {
+      this.clients.forEach(
+        client =>
+          (client.wadoDicomWebClient.headers = this.getAuthorizationHeader())
+      );
+    }
+  }
+
+  // calls reject for each client that supports it
+  public reject() {
+    this.clients.forEach(client => {
+      if (client.supportsReject) {
+        dcm4cheeReject(client.wadoRoot);
```

Comment:
Something doesn't seem right about this reject method. I think it is supposed to return an object with one function named `series` that takes `StudyInstanceUID` and `SeriesInstanceUID` as arguments.